docs/faq/events.md

```diff
@@ -28,3 +30,182 @@ There are lots of times when words just aren't enough to really get clarity on a
+
+### Know your options
+
+If you want to be able to listen for the event on ancestor elements during the "bubble" phase you'll need to manage the `bubbles` property of the event, like so:
+
+```js
+el.dispatchEvent(
+  new Event('event', {
+    bubbles: true,
+  }),
+);
+```
+
+This will allow your event to "bubble" up to the top of the current DOM tree. `bubbles` isn't required in order for an event to be accessible to the ancestors of the element on which is it dispatched, but in that listening to events on the "capture" phase can sometimes be confusing, involving of extended techniques to do in declarative templating, and requiring of extended documentation, it is best that you do use it in these cases.
+
+If you'd like your event to be able to travel across shadow DOM boundaries, and begin its `capture` phase at the `window`, you'll need to manage the event's `composed` property, a la:
+
+```js
+el.dispatchEvent(
+  new Event('event', {
+    composed: true,
+  }),
+);
+```
+
+Be aware that a `composed` event will be retargeted to each shadow root that it passes through. In this way, the value for `event.target` will be the first of an element with a shadow DOM or the dispatching element as you travel from the listening element towards the dispatching element. This is one way in which a shadow root provides encapsulation for the inner working of your element. If you ever need the entire path across the DOM that your event will take you can use the `event.composedPath()` method to acquire an array of elements (starting with the dispatching element) that the event will pass through.
+
+With both of these properties managed, your event will both begin its `capture` phase at the `window` as well as also a `bubble` phase which ends at `window` as well. In this case, your event will look like the following:
+
+```js
+el.dispatchEvent(
+  new Event('event', {
+    bubbles: true,
+    composed: true,
+  }),
+);
+```
+
+Be conscious with your use of `composed: true`. With itall parent components and implementors will also have access to your event in both phases, which can lead to event polution. Only use these settings when you mean for the entire application to have access to an event.
+
+## Even more data
+
+When you'd like to pass data along with your event, you also two options: dispatch a `new CustomEvent()` or extend the `Event` class. While no native DOM element uses the `CustomEvent` constructor to create the events that they dispatch, developers are offered this contructor and its additional `detail` property as a way to pass additional information along with their event. This can be done like so:
```

Comment:
```suggestion
When you'd like to pass data along with your event, you have two options: dispatch a `new CustomEvent()` or extend the `Event` class. While no native DOM element uses the `CustomEvent` constructor to create the events that they dispatch, developers are offered this contructor and its additional `detail` property as a way to pass additional information along with their event. This can be done like so:
```